build.gradle.kts

```diff
@@ -1,130 +1,120 @@
-                        \[Unreleased\]: https:\/\/github\.com\/$repoRef\/compare\/release-$v...HEAD
-                        \[$v\]: https:\/\/github\.com\/$repoRef\/compare\/release-${c.previousVersion}...release-$v$2
+
+                        ## \[$version\] - $currentDateString$1
+                        \[Unreleased\]: https:\/\/github\.com\/$2\/compare\/v$version...HEAD
+                        \[$version\]: https:\/\/github\.com\/$2\/$previousVersionDiffLink$3
                     """.trimIndent()
-                })
-            )
-        )
-        // first-time changelog update--changelog has only unreleased info
-        pre(
-            "fileUpdate",
-            mapOf(
-                "file" to "CHANGELOG.md",
-                "pattern" to KotlinClosure2<String, HookContext, String>({ v, _ ->
-                    "Unreleased([\\s\\S]+?\\nand this project adheres to \\[Semantic Versioning\\]\\(https:\\/\\/semver\\.org\\/spec\\/v2\\.0\\.0\\.html\\).)\\s\\z"
                 }),
-                "replacement" to KotlinClosure2<String, HookContext, String>({ v, c ->
-                    """
-                        \[Unreleased\]
-                        
-                        ## \[$v\] - ${currentDateString()}$1
-                        
-                        \[Unreleased\]: https:\/\/github\.com\/$repoRef\/compare\/release-$v...HEAD
-                        \[$v\]: https:\/\/github\.com\/$repoRef\/releases\/tag\/release-$v
-                    """.trimIndent()
-                })
-            )
+            ),
         )
+
         pre("commit")
-    })
+    }
 }
 
-fun currentDateString() = OffsetDateTime.now(ZoneOffset.UTC).toLocalDate().format(DateTimeFormatter.ISO_DATE)
+group = "org.simplemc"
+version = scmVersion.version
 
-java {
-    sourceCompatibility = JavaVersion.VERSION_11
-    targetCompatibility = JavaVersion.VERSION_11
-}
+val currentDateString: String
+    get() = OffsetDateTime.now(ZoneOffset.UTC).toLocalDate().format(DateTimeFormatter.ISO_DATE)
 
-repositories {
-    jcenter()
-    maven("https://hub.spigotmc.org/nexus/content/repositories/snapshots/")
-    maven("https://oss.sonatype.org/content/repositories/snapshots")
+kotlin {
+    jvmToolchain(21)
 }
 
 dependencies {
-    implementation(kotlin("stdlib-jdk8"))
-    compileOnly(group = "org.spigotmc", name = "spigot-api", version = "$mcApiVersion+")
-}
-
-ktlint {
-    // FIXME - ktlint bug(?): https://github.com/pinterest/ktlint/issues/527
-    disabledRules.set(listOf("import-ordering"))
+    compileOnly(libs.spigot)
+    implementation(libs.kotlinLogger)
 }
 
 tasks {
     wrapper {
-        gradleVersion = "6.1.1"
         distributionType = Wrapper.DistributionType.ALL
     }
 
-    compileKotlin {
-        kotlinOptions {
-            jvmTarget = "1.8"
+    processResources {
+        // inject runtime libraries into online plugin variant
+        val libraries = configurations.runtimeClasspath.get().resolvedConfiguration.resolvedArtifacts
```

Comment:
Injecting `libraries` placeholder is a good candidate to move into the template repo. Without this, any libraries added to the gradle build need to be manually added to the `plugin.yml` `libraries:` list for the non-fat jar distribution to function.